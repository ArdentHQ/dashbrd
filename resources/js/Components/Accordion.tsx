import cn from "classnames";
import React, { useState } from "react";
import { Icon } from "./Icon";
import { isTruthy } from "@/Utils/is-truthy";

interface AccordionItemProperties {
    title: string | JSX.Element;
    children?: JSX.Element;
}

export const AccordionItem = ({ title, children }: AccordionItemProperties): JSX.Element => {
    const [isOpen, setOpen] = useState(false);

    if (!isTruthy(children)) {
        return <HeadOnlyAccordionItem title={title} />;
    }

    return (
        <div
            data-testid="Accordion__item"
            className="select-none"
        >
            <div
                data-testid="Accordion__item-header"
                className="group cursor-pointer"
                onClick={() => {
                    setOpen(!isOpen);
                }}
            >
                <div className="transition-default mx-2 my-1.5 flex min-w-0 items-center rounded-xl p-2 group-hover:bg-theme-secondary-100 dark:group-hover:bg-theme-dark-800 sm:mx-3 sm:p-3">
                    <div className="min-w-0 flex-1">{title}</div>

                    <div className="ml-2 flex h-6 w-6 items-center justify-center">
                        <Icon
                            name="ChevronDownSmall"
                            size="sm"
                            className={cn(
                                "transform text-theme-secondary-900 transition duration-100 dark:text-theme-dark-50",
                                {
                                    "rotate-180": isOpen,
                                },
                            )}
                        />
                    </div>
                </div>
            </div>

            {isOpen && <div className="px-6 pb-4">{children}</div>}
        </div>
    );
};

const HeadOnlyAccordionItem = ({ title }: Pick<AccordionItemProperties, "title">): JSX.Element => (
    <div
        data-testid="HeadOnly_Accordion__item"
        className="select-none"
    >
        <div
            data-testid="HeadOnly_Accordion__item-header"
            className="group pointer-events-none"
        >
            <div className="transition-default mb mx-2 my-1.5 flex min-w-0 items-center rounded-xl p-2 group-hover:bg-theme-secondary-100 sm:mx-3 sm:p-3">
                <div className="min-w-0 flex-1">{title}</div>
            </div>
        </div>
    </div>
);

export const Accordion = ({ children }: { children: React.ReactNode }): JSX.Element => (
    <div
        data-testid="Accordion"
        className="divide-y divide-theme-secondary-300 rounded-xl border border-theme-secondary-300 py-2 dark:divide-theme-dark-700 dark:border-theme-dark-700"
    >
        {children}
    </div>
);
