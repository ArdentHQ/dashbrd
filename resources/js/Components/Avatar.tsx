import Jazzicon, { jsNumberForAddress } from "react-jazzicon";

interface Properties {
    address: string;
    size?: number;
    avatar?: App.Data.Wallet.WalletAvatarData;
}

export const Avatar = ({ address, size = 20, avatar }: Properties): JSX.Element => {
    if (avatar?.default !== null && avatar?.default !== undefined) {
        const sourceSet = [
            avatar.small !== null ? `${avatar.small} 1x` : null,
            avatar.small2x !== null ? `${avatar.small2x} 2x` : null,
        ]
            .filter((value) => value !== null)
            .join(", ");

        return (
            <img
                data-testid="Avatar__image"
                src={avatar.default}
                srcSet={sourceSet}
                className="rounded-full"
                style={{ width: size, height: size }}
            />
        );
    }
    return (
        <Jazzicon
            diameter={size}
            seed={jsNumberForAddress(address)}
        />
    );
};
