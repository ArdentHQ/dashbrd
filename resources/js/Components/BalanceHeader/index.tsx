export * from "./BalanceHeader";
export * from "./BalanceHeaderMobile";
