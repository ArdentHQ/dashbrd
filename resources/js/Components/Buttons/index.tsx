export * from "./Button";
export * from "./LikeButton";
export * from "./IconLikeButton";
export * from "./IconButton";
