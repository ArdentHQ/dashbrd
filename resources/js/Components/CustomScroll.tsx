import cn from "classnames";
import { useCallback, useEffect, useState } from "react";
import { useResizeDetector } from "react-resize-detector";

interface Properties extends React.HTMLAttributes<HTMLDivElement> {}

export const CustomScroll = ({ className, children, ...properties }: Properties): JSX.Element => {
    const [isScroll, setIsScroll] = useState(false);

    const handleResize = useCallback((_width?: number, height?: number) => {
        /* istanbul ignore next -- @preserve */
        setIsScroll((height ?? 0) > window.innerHeight);
    }, []);

    const { height, ref: reference } = useResizeDetector<HTMLDivElement>({
        handleWidth: false,
        onResize: handleResize,
    });

    const handleWindowResize = useCallback(() => {
        handleResize(undefined, height);
    }, [handleResize, height]);

    useEffect(() => {
        window.addEventListener("resize", handleWindowResize);

        return () => {
            window.removeEventListener("resize", handleWindowResize);
        };
    }, [handleWindowResize]);

    return (
        <div
            data-testid="CustomScroll"
            className={cn(
                "custom-scroll relative overflow-x-hidden overflow-y-scroll",
                { "custom-scroll-active": isScroll },
                className,
            )}
            {...properties}
        >
            <div
                ref={reference}
                className={cn({ "-mr-1.5": isScroll })}
            >
                {children}
            </div>
        </div>
    );
};
