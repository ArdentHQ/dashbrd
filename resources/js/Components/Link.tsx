import { Link as InertiaLink } from "@inertiajs/react";
import cn from "classnames";
import { forwardRef, type MouseEvent, useMemo } from "react";
import { Icon } from "@/Components/Icon";
import { useExternalLinkContext } from "@/Contexts/ExternalLinkContext";

interface ClassNameProperties {
    className?: string;
    fontSize?: string;
    textColor?: string;
    variant?: "link";
    disabled?: boolean;
    useAnchorTag?: boolean;
}

interface Properties extends ClassNameProperties {
    external?: boolean;
    href: string;
    children?: React.ReactNode;
    "data-testid"?: string;
    showExternalIcon?: boolean;
    confirmBeforeProceeding?: boolean;
    iconClassName?: string;
}

interface LinkButtonProperties extends Omit<Properties, "href"> {
    onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
}

const variantClassName = ({
    variant,
    className,
    fontSize,
    textColor,
    disabled,
}: ClassNameProperties): string | undefined =>
    useMemo<string | undefined>(() => {
        if (variant === "link") {
            const linkClass = cn(
                "transition-default rounded-full font-medium leading-5.5",
                "underline decoration-transparent underline-offset-2",
                "outline-none outline-3 outline-offset-4",
                {
                    "cursor-not-allowed text-theme-secondary-500 dark:text-theme-dark-200": disabled === true,
                    "hover:text-theme-primary-700 hover:decoration-theme-primary-700 focus-visible:outline-theme-primary-300 dark:hover:text-theme-dark-50 dark:hover:decoration-theme-dark-50 dark:focus-visible:outline-theme-primary-700 dark:focus-visible:text-theme-secondary-700 dark:active:text-theme-dark-100 dark:active:decoration-transparent":
                        disabled !== true,
                },
                className,
            );

            return cn(
                linkClass,
                fontSize ?? "text-sm",
                textColor ?? "text-theme-secondary-700 dark:text-theme-dark-200",
            );
        }

        return className;
    }, [variant, textColor, className]);

const stopPropagationAndBlur = (event: MouseEvent<HTMLElement>): void => {
    event.stopPropagation();
    document.activeElement instanceof HTMLElement && document.activeElement.blur();
};

export const LinkButton = ({
    onClick,
    variant,
    className,
    fontSize,
    textColor,
    ...properties
}: LinkButtonProperties): JSX.Element => (
    <button
        data-testid="Link__button"
        type="button"
        className={variantClassName({ variant, className, fontSize, textColor })}
        onClick={(event: MouseEvent<HTMLButtonElement>) => {
            stopPropagationAndBlur(event);

            onClick?.(event);
        }}
        {...properties}
    />
);

export const Link = forwardRef<HTMLAnchorElement, Properties>(
    (
        {
            external = false,
            disabled = false,
            useAnchorTag = false,
            variant,
            className,
            fontSize,
            textColor,
            showExternalIcon = true,
            confirmBeforeProceeding = false,
            iconClassName,
            children,
            href,
            ...properties
        }: Properties,
        reference,
    ): JSX.Element => {
        const { openConfirmationModal, hasDisabledLinkWarning, isDomainAllowed } = useExternalLinkContext();

        if (external) {
            const handleExternalClick = (event: MouseEvent<HTMLElement>): void => {
                if (confirmBeforeProceeding) {
                    if (!hasDisabledLinkWarning && !isDomainAllowed(href)) {
                        event.preventDefault();
                        event.stopPropagation();

                        openConfirmationModal(href);
                    }
                    return;
                }

                stopPropagationAndBlur(event);
            };

            return (
                <a
                    data-testid="Link__anchor"
                    href={href}
                    target="_blank"
                    ref={reference}
                    rel="noreferrer"
                    className={variantClassName({ variant, className, fontSize, textColor, disabled })}
                    onClick={handleExternalClick}
                    {...properties}
                >
                    {children}

                    {showExternalIcon && (
                        <Icon
                            name="ArrowExternalSmall"
                            size="sm"
                            className={iconClassName ?? "ml-1 text-theme-secondary-500"}
                        />
                    )}
                </a>
            );
        }

        if (useAnchorTag) {
            return (
                <a
                    href={href}
                    ref={reference}
                    className={variantClassName({ variant, className, fontSize, textColor, disabled })}
                    {...properties}
                    data-testid="Link__anchor"
                >
                    {children}
                </a>
            );
        }

        return (
            <InertiaLink
                {...properties}
                href={href}
                disabled={disabled}
                className={variantClassName({ variant, className, fontSize, textColor, disabled })}
            >
                {children}
            </InertiaLink>
        );
    },
);

Link.displayName = "Link";
