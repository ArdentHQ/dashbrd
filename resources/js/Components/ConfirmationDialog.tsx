import cn from "classnames";
import { type FormEvent, type RefObject, useRef } from "react";
import { Button, type ButtonVariant } from "./Buttons/Button";
import { Dialog } from "@/Components/Dialog";

interface Properties {
    title: string;
    cancelLabel?: string;
    confirmLabel: string;
    isOpen: boolean;
    isDisabled?: boolean;
    isStatic?: boolean;
    onClose: () => void;
    onCancel?: () => void;
    onConfirm: () => void;
    children: React.ReactNode;
    focus?: RefObject<HTMLElement>;
    hasBlurryOverlay?: boolean;
    childrenClassName?: string;
    confirmationButtonVariant?: ButtonVariant;
}

export const ConfirmationDialog = ({
    title,
    cancelLabel = "Cancel",
    confirmLabel,
    isOpen,
    isDisabled = false,
    isStatic = false,
    onClose,
    onConfirm,
    onCancel,
    children,
    focus,
    hasBlurryOverlay = false,
    childrenClassName,
    confirmationButtonVariant = "primary",
}: Properties): JSX.Element => {
    const submitButton = useRef<HTMLButtonElement>(null);

    const onSubmit = (event: FormEvent): void => {
        event.preventDefault();
        event.stopPropagation();

        onConfirm();
    };

    return (
        <Dialog
            title={title}
            isStatic={isStatic}
            isOpen={isOpen}
            onClose={onClose}
            focus={focus ?? submitButton}
            isUsedByConfirmationDialog
            hasBlurryOverlay={hasBlurryOverlay}
        >
            <form
                data-testid="ConfirmationDialog__form"
                className="flex min-h-full flex-col"
                onSubmit={onSubmit}
            >
                <div className={cn("flex-1 px-6 pb-6 pt-4", childrenClassName)}>{children}</div>

                <footer className="flex items-center justify-end space-x-3 border-t border-theme-secondary-300 px-6 py-4 dark:border-theme-dark-700 sm:px-6 sm:pb-6 sm:pt-4">
                    {!isStatic && (
                        <Button
                            data-testid="ConfirmationDialog__close"
                            variant="secondary"
                            className="w-full justify-center sm:w-auto"
                            onClick={onCancel ?? onClose}
                        >
                            {cancelLabel}
                        </Button>
                    )}

                    <Button
                        data-testid="ConfirmationDialog__confirm"
                        ref={submitButton}
                        type="submit"
                        processing={isDisabled}
                        disabled={isDisabled}
                        className="w-full justify-center sm:w-auto"
                        variant={confirmationButtonVariant}
                    >
                        {confirmLabel}
                    </Button>
                </footer>
            </form>
        </Dialog>
    );
};
