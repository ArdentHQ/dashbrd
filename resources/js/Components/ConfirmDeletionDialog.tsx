import { useMemo, useRef, useState } from "react";
import { useTranslation } from "react-i18next";
import { type ButtonVariant } from "./Buttons";
import { ConfirmationDialog } from "@/Components/ConfirmationDialog";
import { TextInput } from "@/Components/Form/TextInput";
import { DeleteModal } from "@/images";

interface Properties {
    isOpen: boolean;
    onClose: () => void;
    onConfirm: () => void;
    title: string;
    children: React.ReactNode;
    isDisabled?: boolean;
    requiresConfirmation?: boolean;
    confirmationButtonVariant?: ButtonVariant;
}

export const ConfirmDeletionDialog = ({
    isOpen,
    isDisabled = false,
    onClose,
    onConfirm,
    children,
    title,
    requiresConfirmation = true,
    confirmationButtonVariant,
}: Properties): JSX.Element => {
    const { t } = useTranslation();

    const input = useRef<HTMLInputElement>(null);
    const [confirmationValue, setConfirmationValue] = useState("");

    const valid = useMemo(
        () => !requiresConfirmation || confirmationValue.toLowerCase() === t("common.delete").toLowerCase(),
        [confirmationValue, requiresConfirmation],
    );

    return (
        <ConfirmationDialog
            title={title}
            isOpen={isOpen}
            confirmLabel={t("common.confirm")}
            onConfirm={onConfirm}
            isDisabled={isDisabled || !valid}
            onClose={() => {
                setConfirmationValue("");
                onClose();
            }}
            focus={input}
            confirmationButtonVariant={confirmationButtonVariant}
        >
            <div className="space-y-3">
                <DeleteModal className="mx-auto" />

                <p className="text-theme-secondary-700 dark:text-theme-dark-200">{children}</p>

                {requiresConfirmation && (
                    <TextInput
                        ref={input}
                        onChange={(event) => {
                            setConfirmationValue(event.target.value);
                        }}
                        data-testid="ConfirmDeletionDialog__input"
                        placeholder={t("common.write_to_confirm", {
                            word: t("common.delete").toUpperCase(),
                        })}
                    />
                )}
            </div>
        </ConfirmationDialog>
    );
};
