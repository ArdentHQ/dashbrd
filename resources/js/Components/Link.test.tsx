import React from "react";
import { expect } from "vitest";
import { Link, LinkButton } from "@/Components/Link";
import { ExternalLinkContextProvider } from "@/Contexts/ExternalLinkContext";
import { render, screen, userEvent } from "@/Tests/testing-library";

describe("Link", () => {
    it("should render", () => {
        const { asFragment } = render(
            <Link href="/test-link">
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("test")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render as external", () => {
        const { asFragment } = render(
            <Link
                href="/test-link"
                external
                iconClassName="text-base"
            >
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("Link__anchor")).toBeInTheDocument();
        expect(screen.getByTestId("icon-ArrowExternalSmall")).toBeInTheDocument();
        expect(screen.getByTestId("icon-ArrowExternalSmall")).toHaveClass("text-base");
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render with link variant classname", () => {
        const { asFragment } = render(
            <Link
                href="/test-link"
                external
                variant="link"
            >
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("Link__anchor")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render with a link without an external icon", () => {
        render(
            <Link
                href="/test-link"
                external
                variant="link"
                showExternalIcon={false}
            >
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("Link__anchor")).toBeInTheDocument();
        expect(screen.queryByTestId("icon-ArrowExternalSmall")).not.toBeInTheDocument();
    });

    it("should render with custom text color", () => {
        render(
            <Link
                href="/test-link"
                external
                textColor="text-theme-primary-600"
                variant="link"
            >
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("Link__anchor")).toBeInTheDocument();
        expect(screen.getByTestId("Link__anchor")).toHaveClass("text-theme-primary-600");
    });
});

describe("LinkButton", () => {
    it("should render", () => {
        const { asFragment } = render(
            <LinkButton onClick={() => vi.fn()}>
                <span data-testid="test"></span>
            </LinkButton>,
        );

        expect(screen.getByTestId("test")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render with link variant classname", () => {
        const { asFragment } = render(
            <LinkButton
                onClick={() => vi.fn()}
                variant="link"
            >
                <span data-testid="test"></span>
            </LinkButton>,
        );

        expect(screen.getByTestId("Link__button")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render as external with confirm", () => {
        render(
            <Link
                href="https://duhverse.com"
                external
                confirmBeforeProceeding
                iconClassName="text-base"
            >
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("Link__anchor")).toBeInTheDocument();
        expect(screen.getByTestId("icon-ArrowExternalSmall")).toBeInTheDocument();
        expect(screen.getByTestId("icon-ArrowExternalSmall")).toHaveClass("text-base");
    });

    it("should open and close external confirm modal", async () => {
        render(
            <ExternalLinkContextProvider allowedExternalDomains={[]}>
                <Link
                    href="https://test.com"
                    external
                    confirmBeforeProceeding
                >
                    <span data-testid="test"></span>
                </Link>
            </ExternalLinkContextProvider>,
        );

        await userEvent.click(screen.getByTestId("Link__anchor"));

        expect(screen.getByTestId("Toast")).toBeInTheDocument();
        expect(screen.getByTestId("ExternalLinkConfirmModal__info")).toBeInTheDocument();

        await userEvent.click(screen.getByTestId("ConfirmationDialog__close"));

        expect(screen.queryByTestId("Toast")).not.toBeInTheDocument();
        expect(screen.queryByTestId("ExternalLinkConfirmModal__info")).not.toBeInTheDocument();
    });

    it("should not open external confirm modal if user disabled it previously", async () => {
        window.localStorage.setItem("has_disabled_link_warning", "true");

        render(
            <ExternalLinkContextProvider allowedExternalDomains={["test.com"]}>
                <Link
                    href="https://test.com"
                    external
                    confirmBeforeProceeding
                >
                    <span data-testid="test"></span>
                </Link>
            </ExternalLinkContextProvider>,
        );

        await userEvent.click(screen.getByTestId("Link__anchor"));

        expect(screen.queryByTestId("ExternalLinkConfirmModal__info")).not.toBeInTheDocument();
        window.localStorage.setItem("has_disabled_link_warning", "false");
    });

    it("should not open modal if domain is not allowed", async () => {
        render(
            <ExternalLinkContextProvider allowedExternalDomains={[]}>
                <Link
                    href="https://test.com"
                    external
                    confirmBeforeProceeding
                >
                    <span data-testid="test"></span>
                </Link>
            </ExternalLinkContextProvider>,
        );

        await userEvent.click(screen.getByTestId("Link__anchor"));

        expect(screen.getByTestId("Toast")).toBeInTheDocument();
        expect(screen.getByTestId("ExternalLinkConfirmModal__info")).toBeInTheDocument();

        await userEvent.click(screen.getByTestId("ConfirmationDialog__close"));

        expect(screen.queryByTestId("Toast")).not.toBeInTheDocument();
        expect(screen.queryByTestId("ExternalLinkConfirmModal__info")).not.toBeInTheDocument();
    });

    it("should not open confirmation link if confirmBeforeProceeding is not true", async () => {
        render(
            <ExternalLinkContextProvider allowedExternalDomains={[]}>
                <Link
                    href="https://test.com"
                    external
                >
                    <span data-testid="test"></span>
                </Link>
            </ExternalLinkContextProvider>,
        );

        await userEvent.click(screen.getByTestId("Link__anchor"));

        expect(screen.queryByTestId("Toast")).not.toBeInTheDocument();
        expect(screen.queryByTestId("ExternalLinkConfirmModal__info")).not.toBeInTheDocument();
    });

    it("should avoid external confirm modal", async () => {
        render(
            <Link
                href="https://duhverse.com"
                external
                confirmBeforeProceeding
            >
                <span data-testid="test"></span>
            </Link>,
        );

        await userEvent.click(screen.getByTestId("Link__anchor"));

        expect(screen.queryByTestId("Toast")).not.toBeInTheDocument();
        expect(screen.queryByTestId("ExternalLinkConfirmModal__info")).not.toBeInTheDocument();
    });

    it("should render an anchor if href is a link to a section", () => {
        render(
            <Link
                href="#test"
                useAnchorTag
                confirmBeforeProceeding
            >
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("Link__anchor")).toBeInTheDocument();
    });

    it("should render an anchor if useAnchorTag is true", () => {
        render(
            <Link
                href="#test"
                confirmBeforeProceeding
                useAnchorTag
            >
                <span data-testid="test"></span>
            </Link>,
        );

        expect(screen.getByTestId("Link__anchor")).toBeInTheDocument();
    });
});
