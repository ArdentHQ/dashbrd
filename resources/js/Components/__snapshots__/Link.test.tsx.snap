// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > should render 1`] = `
<DocumentFragment>
  <a
    href="/test-link"
  >
    <span
      data-testid="test"
    />
  </a>
</DocumentFragment>
`;

exports[`Link > should render as external 1`] = `
<DocumentFragment>
  <a
    data-testid="Link__anchor"
    href="/test-link"
    rel="noreferrer"
    target="_blank"
  >
    <span
      data-testid="test"
    />
    <svg
      class="text-base"
      data-testid="icon-ArrowExternalSmall"
      fill="none"
      height="12"
      viewBox="0 0 12 12"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.4 3.9V.6H8.1m3.3 0L4.2 7.8m1.7-5.4H1.4c-.4 0-.8.4-.8.7v7.5c0 .4.4.8.7.8h7.5c.4 0 .7-.4.7-.8V6.1"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.2"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`Link > should render with link variant classname 1`] = `
<DocumentFragment>
  <a
    class="transition-default rounded-full font-medium leading-5.5 underline decoration-transparent underline-offset-2 outline-none outline-3 outline-offset-4 hover:text-theme-primary-700 hover:decoration-theme-primary-700 focus-visible:outline-theme-primary-300 dark:hover:text-theme-dark-50 dark:hover:decoration-theme-dark-50 dark:focus-visible:outline-theme-primary-700 dark:focus-visible:text-theme-secondary-700 dark:active:text-theme-dark-100 dark:active:decoration-transparent text-sm text-theme-secondary-700 dark:text-theme-dark-200"
    data-testid="Link__anchor"
    href="/test-link"
    rel="noreferrer"
    target="_blank"
  >
    <span
      data-testid="test"
    />
    <svg
      class="ml-1 text-theme-secondary-500"
      data-testid="icon-ArrowExternalSmall"
      fill="none"
      height="12"
      viewBox="0 0 12 12"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.4 3.9V.6H8.1m3.3 0L4.2 7.8m1.7-5.4H1.4c-.4 0-.8.4-.8.7v7.5c0 .4.4.8.7.8h7.5c.4 0 .7-.4.7-.8V6.1"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.2"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`LinkButton > should render 1`] = `
<DocumentFragment>
  <button
    data-testid="Link__button"
    type="button"
  >
    <span
      data-testid="test"
    />
  </button>
</DocumentFragment>
`;

exports[`LinkButton > should render with link variant classname 1`] = `
<DocumentFragment>
  <button
    class="transition-default rounded-full font-medium leading-5.5 underline decoration-transparent underline-offset-2 outline-none outline-3 outline-offset-4 hover:text-theme-primary-700 hover:decoration-theme-primary-700 focus-visible:outline-theme-primary-300 dark:hover:text-theme-dark-50 dark:hover:decoration-theme-dark-50 dark:focus-visible:outline-theme-primary-700 dark:focus-visible:text-theme-secondary-700 dark:active:text-theme-dark-100 dark:active:decoration-transparent text-sm text-theme-secondary-700 dark:text-theme-dark-200"
    data-testid="Link__button"
    type="button"
  >
    <span
      data-testid="test"
    />
  </button>
</DocumentFragment>
`;
